apps/etterlatte-saksbehandling-ui/client/src/components/person/notat/NotatRedigeringModal.tsx

```diff
@@ -0,0 +1,168 @@
+            <Tabs.List>
+              <Tabs.Tab
+                value={NotatRedigeringFane.REDIGER}
+                label="Rediger"
+                icon={<PencilIcon fontSize="1.5rem" aria-hidden />}
+              />
+              <Tabs.Tab
+                value={NotatRedigeringFane.FORHAANDSVIS}
+                label="Forhåndsvisning"
+                icon={<FilePdfIcon fontSize="1.5rem" aria-hidden />}
+              />
+            </Tabs.List>
+
+            <Tabs.Panel value={NotatRedigeringFane.REDIGER}>
+              <Box paddingBlock="4">
+                <VStack gap="4">
+                  <RedigerNotatTittel id={notat.id} tittel={notat.tittel} />
+
+                  <Heading size="xsmall">Rediger notat</Heading>
+
+                  {mapResult(hentPayloadStatus, {
+                    pending: <Spinner label="Henter notat ..." visible />,
+                    success: () => (
+                      <SlateEditor value={content} onChange={(value) => setContent(value)} readonly={false} />
+                    ),
+                  })}
+                </VStack>
+              </Box>
+            </Tabs.Panel>
+
+            <Tabs.Panel value={NotatRedigeringFane.FORHAANDSVIS}>
+              <ForhaandsvisningNotat id={notat.id} />
+            </Tabs.Panel>
+          </Tabs>
+        </Modal.Body>
+
+        <Modal.Footer>
+          {isFailureHandler({
+            apiResult: lagrePayloadStatus,
+            errorMessage: 'Feil oppsto ved lagring av notat',
+          })}
+
+          {toggleJournalfoer ? (
+            <VStack gap="4">
+              <Alert variant="warning">
+                Er du sikker på at du vil journalføre notatet? Handlingen kan ikke angres.
+              </Alert>
+
+              <HStack gap="4" align="center">
+                <Button variant="tertiary" onClick={() => setToggleJournalfoer(false)}>
+                  Nei, avbryt
+                </Button>
+                <Button onClick={journalfoer} loading={isPending(journalfoerStatus)}>
+                  Ja, journalfør
+                </Button>
+              </HStack>
+            </VStack>
+          ) : (
+            <HStack gap="4" align="center">
+              {isSuccess(lagrePayloadStatus) && lagretStatus.beskrivelse && (
+                <Detail as="span">
+                  <Alert variant="success" size="small" inline>
+                    {lagretStatus.beskrivelse}
+                  </Alert>
+                </Detail>
+              )}
+              <Button variant="tertiary" onClick={() => setIsOpen(false)}>
+                Lukk
+              </Button>
+              <Button
+                onClick={() => {
+                  setFane(NotatRedigeringFane.FORHAANDSVIS)
+                  setToggleJournalfoer(true)
+                }}
+              >
+                Journalfør
+              </Button>
+            </HStack>
+          )}
+        </Modal.Footer>
+      </DokumentVisningModal>
```

Comment:
Mulighet for at noe av dette kanskje kan splittes ut? IMO skjer det litt mye i samme komponent